src/arduino_lmic_hal_configuration.h

```diff
@@ -70,7 +70,7 @@ class HalConfiguration_t
 public:
 	HalConfiguration_t() {};
 
-	virtual ostime_t setModuleActive(bool state)
+	virtual ostime_t setModuleActive(bool __attribute__((unused)) state)
```

Comment:
the convention I've been using (to avoid GCC dependencies) is to use `LMIC_API_PARAMETER()`:

https://github.com/mcci-catena/arduino-lmic/blob/d9bb0f25da0de55e4549a0e133b1a90d834bd422/src/lmic/lmic_env.h#L71-L92

So you'd instead add `LMIC_API_PARAMETER(state);` at line 75.